packages/backend/src/submissions/submissions.service.ts

```diff
@@ -1,29 +1,60 @@
-import { Injectable } from '@nestjs/common'
+import { BadRequestException, Injectable, NotFoundException } from '@nestjs/common'
+import { Prisma } from '@prisma/client'
+import { PrismaService } from 'nestjs-prisma'
 import { CreateSubmissionDto } from './dto/create-submission.dto'
 import { UpdateSubmissionDto } from './dto/update-submission.dto'
-import { PrismaService } from 'nestjs-prisma'
+import { SubmissionEntitiy } from './entities/submission.entity'
 
 @Injectable()
 export class SubmissionsService {
   constructor(private prisma: PrismaService) {}
 
-  create(createSubmissionDto: CreateSubmissionDto) {
-    return 'This action adds a new submission'
+  async create(data: CreateSubmissionDto): Promise<SubmissionEntitiy> {
+    try {
+      return await this.prisma.submission.create({ data })
+    } catch (e) {
+      if (e instanceof Prisma.PrismaClientKnownRequestError) {
+        if (e.code === 'P2002') {
+          throw new BadRequestException('Unique constraint violation')
+        }
+        throw e
+      }
+    }
```

Comment:
This handles the violation of the unique constraint correctly, but another Prisma error can occur.
If you submit a submission with an `oldLinkId` that doesn't exist in the `Link` table, the foreign key constraint will be violated. I believe the code of that error is `P2003`, it should be handled the same way as `P2002`